### ISSUE TITLE
Continuing difficult with auth to private dockerhub repository

### ISSUE THREAD

**Issue by Badg** [terraform-providers/terraform-provider-docker#125]
This looks very related to #4, but it seems like #21 should have fixed this. However, I'm still unable to use the docker config file to auth, and instead needed to manually add in credentials to the .tf. My hunch is that it's related to docker desktop now using the osx keychain instead of keeping the creds in "plaintext" (well, base64, but, yeah, that's plaintext) in ~/.docker/config.json. As a result, I'm honestly not sure if this is going to end up being a bug or a feature request.

### Terraform Version
```
Terraform v0.11.11
+ provider.docker v1.1.0
```

### Affected Resource(s)

+ ``registry_auth``? ``provider "docker"``? Not really sure.

### Terraform Configuration Files

What didn't work:

```hcl
provider "docker" {
  host     = "unix:///var/run/docker.sock"

  registry_auth {
    address = "https://registry.hub.docker.com"
    config_file = "~/.docker/config.json"
  }
}
...
```

I also tried omitting the registry auth entirely. What did finally work:

```hcl
variable "dockerhub_username" {}
variable "dockerhub_pass" {}

provider "docker" {
  host     = "unix:///var/run/docker.sock"

  registry_auth {
    address = "https://registry.hub.docker.com"
    username = "${var.dockerhub_username}"
    password = "${var.dockerhub_pass}"
  }
}
```

I'm willing to bet that the problem  here is that newer versions of docker are using credential helpers instead of directly encoding the auth for it. For example, here's a relevant snippet from my docker config:

```json
{
	"auths": {
		"https://index.docker.io/v1/": {},
		"https://registry.hub.docker.com": {}
	},
	"credsStore": "osxkeychain"
}
```

That is **not redacted**, because no redaction is necessary! Docker knows to go into my keychain and grab those credentials there.

### Debug Output

There are two [here](https://gist.github.com/Badg/6c6ebc5ceb914c91374e9f8239117b36). The first (failed to pull) is when I leave out the ``registry_auth`` block, and the second (failed to find registry config) is when I point it to the docker config file.

### Panic Output

N/A

### Expected Behavior

Auth should have been loaded and the pull should have succeeded!

### Actual Behavior

Either auth fails during pulling, or it  fails to load the configfile, erroring out immediately after starting ``apply``.

### Steps to Reproduce

For me, this happened automatically:

1. have private repo on dockerhub
2. `terraform apply` with something using an image from it

But I suspect the critical component is:
0. Docker desktop uses the osx keychain for credential management now

This is true even when I can immediately do an identical command using ``docker pull <foo>``.

### Important Factoids

See above re: osx keychain, and also that docker pull works.

### References

+ #4 
+ #21

**Comment by mavogel** [comment]
Hi,
thanks for pointing this out. I can reproduce your error.
I will probably examine how the `docker cli` accesses the osxkeychain and provide a solution for this. AFAIK on Linux the base64 encoded credentials are still in the `~/.docker/config.json`, that's why the tests on travis work fine. I'll also take a look at https://github.com/keybase/go-keychain

**Comment by mavogel** [comment]
Closed in #143
